ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h

```diff
@@ -216,6 +214,13 @@ void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
         auto& eps = _ip_data[ip].eps;
         auto const& sigma_eff = _ip_data[ip].sigma_eff;
 
+        auto const T_ref =
+            medium->hasProperty(MPL::PropertyType::reference_temperature)
+                ? medium->property(MPL::PropertyType::reference_temperature)
+                      .template value<double>(vars, x_position, t)
+                : std::numeric_limits<double>::quiet_NaN();
+
```

Comment:
Does somebody have a better idea?
I could make T_ref a required property, but since it is not needed in most of the cases I don't think that makes sense

Reply:
Is T_ref independent from the integration point? Then it could be evaluated only once outside of the integration loop.
If T_ref is additionally independent from time it could be evaluated in the constructor and stored in the local assembler. Or if it is even independent from the mesh element? Then, is it possible to move it out of the class and pass it as an argument to the local assembler?

Reply:
Since the user could in theory make t_ref spatially dependent (e.g. cavern vertical temperature gradient) it is dependent of the IP, right?
I think with the time dependence it's the same. 
I could move it out of the loop when we assume it is constant per element.
But since it is probably a rare scenario having something else than a constant T_ref in a HM-Process, we maybe can move it further out. But then we could just leave it as a variable in process_data...

Reply:
Consensus is, to make T_ref a required property